devdoc/tqueue_requirements.md

```diff
@@ -241,39 +215,65 @@ TQUEUE_PUSH_RESULT TQUEUE_PUSH(T)(TQUEUE(T) tqueue, T* item, void* copy_item_fun
+
+  - **SRS_TQUEUE_01_061: [** If the current queue size is equal to the max queue size, `TQUEUE_PUSH(T)` shall return `TQUEUE_PUSH_QUEUE_FULL`. **]**
 
-- If the queue is full (current head >= current tail + queue size):
+  - **SRS_TQUEUE_01_062: [** If the current queue size is less than the max queue size: **]**
 
-  - If the queue is not growable, `TQUEUE_PUSH(T)` shall return `TQUEUE_PUSH_QUEUE_FULL`.
+    - **SRS_TQUEUE_01_063: [** `TQUEUE_PUSH(T)` shall release in shared mode the lock used to guard the growing of the queue. **]**
 
-  - If the queue is growable:
+    - **SRS_TQUEUE_01_064: [** `TQUEUE_PUSH(T)` shall acquire in exclusive mode the lock used to guard the growing of the queue. **]**
 
-    - `TQUEUE_PUSH(T)` shall release in shared mode the lock used to guard the growing of the queue.
+    - **SRS_TQUEUE_01_074: [** If the size of the queue did not change after acquiring the lock in shared mode: **]**
 
-    - `TQUEUE_PUSH(T)` shall acquire in exclusive mode the lock used to guard the growing of the queue.
+      - **SRS_TQUEUE_01_075: [** `TQUEUE_PUSH(T)` shall obtain again the current head or the queue. **]**
 
-    - If the size of the queue did not change after acquiring the lock in shared mode:
+      - **SRS_TQUEUE_01_076: [** `TQUEUE_PUSH(T)` shall obtain again the current tail or the queue. **]**
     
-      - `TQUEUE_PUSH(T)` shall double the size of the queue.
+      - **SRS_TQUEUE_01_067: [** `TQUEUE_PUSH(T)` shall double the size of the queue. **]**
 
-      - If the newly computed queue size is higher than the `max_queue_size` value passed to `TQUEUE_CREATE_GROWABLE(T)`, `TQUEUE_PUSH(T)` shall use `max_queue_size` as the new queue size.
+      - **SRS_TQUEUE_01_070: [** If the newly computed queue size is higher than the `max_queue_size` value passed to `TQUEUE_CREATE(T)`, `TQUEUE_PUSH(T)` shall use `max_queue_size` as the new queue size. **]**
 
-      - `TQUEUE_PUSH(T)` shall reallocate the array used to store the queue items based on the newly computed size.
+      - **SRS_TQUEUE_01_068: [** `TQUEUE_PUSH(T)` shall reallocate the array used to store the queue items based on the newly computed size. **]**
 
-      - If reallocation fails, `TQUEUE_PUSH(T)` shall return `TQUEUE_PUSH_ERROR`.
+      - **SRS_TQUEUE_01_077: [** `TQUEUE_PUSH(T)` shall move the entries between the tail index and the array end like below: **]**
 
-    - Otherwise, `TQUEUE_PUSH(T)` shall release in exclusive mode the lock used to guard the growing of the queue.
+        - **SRS_TQUEUE_01_078: [** Entries at the tail shall be moved to the end of the resized array
 
-    - `TQUEUE_PUSH(T)` shall acquire in shared mode the lock used to guard the growing of the queue and retry the `TQUEUE_PUSH(T)`.
+      Before resize:
+
+      T = 2
+      H = 5
+
+      [X HO TX X]
+
+      After resize (doubling from 4 to 8):
+
+      T = 6
+      H = 9
+
+      [X HO O O O O TX X] **]**
```

Comment:
>**]** [](http://example.com/codeflow?start=26&length=5)

nit: maybe should just be on line 250 and this "diagram" just exists here outside of the SRS line

Reply:
Updated it to have a comment in the to look at the spec and changed the spec to not actually include the diagram.